### PR TITLE
[formrecognizer] update readme

### PR DESCRIPTION
**Pull request by kristapratico**
Resolves https://github.com/Azure/azure-sdk-for-python/issues/20846

TODO: needs migration guide link, needs prebuilt-document example

### CODE REVIEW

File: sdk/formrecognizer/azure-ai-formrecognizer/README.md [hunk 1]
```diff
@@ -119,42 +128,44 @@ Once completed, set the values of the client ID, tenant ID, and client secret of
...
+|"{custom-model-id}" | Text extraction, selection marks, tables, labeled fields and values from your custom documents
+|"prebuilt-layout" | Text extraction, selection marks, tables
+|"prebuilt-document" | Text extraction, selection marks, tables, key-value pairs and entities
+|"prebuilt-invoices" | Text extraction, selection marks, tables, and pre-trained fields and values pertaining to invoices
+|"prebuilt-businessCard" | Text extraction and pre-trained fields and values pertaining to business cards
+|"prebuilt-idDocument" | Text extraction and pre-trained fields and values pertaining to identity documents
+|"prebuilt-receipt" | Text extraction and pre-trained fields and values pertaining to US sales receipts
 
-Sample code snippets are provided to illustrate using a FormRecognizerClient [here](#recognize-forms-using-a-custom-model "Recognize Forms Using a Custom Model").
+Sample code snippets are provided to illustrate using a DocumentAnalysisClient [here](#analyze-documents-using-a-custom-model "Analyze Documents Using a Custom Model").
```
Comment (catalinaperalta) [inline sdk/formrecognizer/azure-ai-formrecognizer/README.md:156]: Just wondering whether this link should point to the general samples folder? The way it is worded, made me thing that that's what was going to be linked. Or if not, maybe reword to say something like:
```suggestion
Sample code snippets are provided to illustrate using a DocumentAnalysisClient with a custom model [here](#analyze-documents-using-a-custom-model "Analyze Documents Using a Custom Model").
```

File: sdk/formrecognizer/azure-ai-formrecognizer/README.md [hunk 2]
```diff
@@ -171,112 +182,184 @@ Sample code snippets are provided to illustrate using long-running operations [b
...
+from azure.ai.formrecognizer import DocumentAnalysisClient
 from azure.core.credentials import AzureKeyCredential
 
-endpoint = "https://<region>.api.cognitive.microsoft.com/"
+endpoint = "https://<my-custom-subdomain>.cognitiveservices.azure.com/"
 credential = AzureKeyCredential("<api_key>")
 
-form_recognizer_client = FormRecognizerClient(endpoint, credential)
+document_analysis_client = DocumentAnalysisClient(endpoint, credential)
 model_id = "<your custom model id>"
```
Comment (catalinaperalta) [inline sdk/formrecognizer/azure-ai-formrecognizer/README.md:203]: Nit?
```suggestion
model = "<your custom model id>"
```

File: sdk/formrecognizer/azure-ai-formrecognizer/README.md [hunk 3]
```diff
@@ -171,112 +182,184 @@ Sample code snippets are provided to illustrate using long-running operations [b
...
+                "...Selection mark is '{}' and has a confidence of {}".format(
+                    selection_mark.state, selection_mark.confidence
+                )
+            )
+
+# iterate over tables in document
+for i, table in enumerate(result.tables):
+    print("\nTable {} can be found on page:".format(i + 1))
+    for region in table.bounding_regions:
+        print("...{}".format(i + 1, region.page_number))
```
Comment (catalinaperalta) [inline sdk/formrecognizer/azure-ai-formrecognizer/README.md:242]: Is this the right code now? I think it should be clearer that bounding region returns the page number and the corresponding bounding box?

File: sdk/formrecognizer/azure-ai-formrecognizer/README.md [hunk 4]
```diff
@@ -171,112 +182,184 @@ Sample code snippets are provided to illustrate using long-running operations [b
...
 
-Alternatively, a form URL can also be used to recognize custom forms using the `begin_recognize_custom_forms_from_url` method.
-The `_from_url` methods exist for all the recognize methods.
+Alternatively, a document URL can also be used to analyze documents using the `begin_analyze_document_from_url` method.
 
 ```
-form_url = "<url_of_the_form>"
-poller = form_recognizer_client.begin_recognize_custom_forms_from_url(model_id=model_id, form_url=form_url)
+document_url = "<url_of_the_document>"
+poller = document_analysis_client.begin_analyze_document_from_url(model_id=model_id, document_url=document_url)
```
Comment (catalinaperalta) [inline sdk/formrecognizer/azure-ai-formrecognizer/README.md:255]:
```suggestion
poller = document_analysis_client.begin_analyze_document_from_url(model=model_id, document_url=document_url)
```

File: sdk/formrecognizer/azure-ai-formrecognizer/README.md [hunk 5]
```diff
@@ -171,112 +182,184 @@ Sample code snippets are provided to illustrate using long-running operations [b
...
+            )
+
 ```
 
 ### Using Prebuilt Models
-Extract fields from certain types of common forms such as receipts, invoices, business cards, and identity documents using prebuilt models provided by the Form Recognizer service.
+Extract fields from certain types of common forms such as receipts, invoices, business cards, identity documents, and general documents using prebuilt models provided by the Form Recognizer service.
 
-For example, to extract fields from a sales receipt, use the prebuilt receipt model provided by the `begin_recognize_receipts` method:
+For example, to extract fields from a sales receipt, use the prebuilt receipt model provided by passing `model_id="prebuilt-receipt"` into the `begin_analyze_documents` method:
```
Comment (catalinaperalta) [inline sdk/formrecognizer/azure-ai-formrecognizer/README.md:345]:
```suggestion
For example, to extract fields from a sales receipt, use the prebuilt receipt model provided by passing `model="prebuilt-receipt"` into the `begin_analyze_documents` method:
```

File: sdk/formrecognizer/azure-ai-formrecognizer/README.md [hunk 6]
```diff
@@ -119,42 +128,44 @@ Once completed, set the values of the client ID, tenant ID, and client secret of
...
- - Recognizing common fields from the following form types using prebuilt models. These fields and metadata are returned in a collection of `RecognizedForm` objects.
-    - Sales receipts. See fields found on a receipt [here][service_recognize_receipt].
-    - Business cards. See fields found on a business card [here][service_recognize_business_cards].
-    - Invoices. See fields found on an invoice [here][service_recognize_invoice].
-    - Identity documents. See fields found on identity documents [here][service_recognize_identity_documents].
- - Recognizing form content, including tables, lines, words, and selection marks, without the need to train a model. Form content is returned in a collection of `FormPage` objects.
+|Model ID| Features
+|-|-
+|"{custom-model-id}" | Text extraction, selection marks, tables, labeled fields and values from your custom documents
+|"prebuilt-layout" | Text extraction, selection marks, tables
```
[the remark this answers is not in the document]
Comment (kristapratico) [reply sdk/formrecognizer/azure-ai-formrecognizer/README.md:151]: ordered based on Paul's feedback

File: sdk/formrecognizer/azure-ai-formrecognizer/README.md [hunk 7]
```diff
@@ -119,42 +128,44 @@ Once completed, set the values of the client ID, tenant ID, and client secret of
...
 )
 ```
 
 ## Key concepts
 
-### FormRecognizerClient
-`FormRecognizerClient` provides operations for:
+### DocumentAnalysisClient
+`DocumentAnalysisClient` provides operations for analyzing input documents using custom and prebuilt models through the `begin_analyze_document` and `begin_analyze_document_from_url` APIs.
+Use the `model_id` parameter to select the type of model for analysis.
```
[the remark this answers is not in the document]
Comment (kristapratico) [reply sdk/formrecognizer/azure-ai-formrecognizer/README.md:146]: good catch, will fix all refs

File: sdk/formrecognizer/azure-ai-formrecognizer/README.md (hunk 1 above)
Comment (kristapratico) [reply sdk/formrecognizer/azure-ai-formrecognizer/README.md:159]: this was added because in ux studies people wouldn't continue to scroll down on the readme and see the examples. I'm going to update both the DAC and DMAC to link to the examples header.

File: sdk/formrecognizer/azure-ai-formrecognizer/README.md (hunk 2 above)
Comment (kristapratico) [reply sdk/formrecognizer/azure-ai-formrecognizer/README.md]: This model_id maps to the one from build so I think we should keep it.

File: sdk/formrecognizer/azure-ai-formrecognizer/README.md (hunk 3 above)
Comment (kristapratico) [reply sdk/formrecognizer/azure-ai-formrecognizer/README.md]: I updated based on Paul's recommendations. Will try running these snippets shortly to ensure accuracy

File: sdk/formrecognizer/azure-ai-formrecognizer/README.md [hunk 8]
```diff
@@ -1,15 +1,12 @@
...
+It includes the following main features:
 
-* Custom models - Recognize field values and table data from forms. These models are trained with your own data, so they're tailored to your forms.
-* Content API - Recognize text, table structures, and selection marks, along with their bounding box coordinates, from documents. Corresponds to the REST service's Layout API.
-* Prebuilt models - Recognize data using the following prebuilt models
-    * Receipt model - Recognize data from sales receipts using a prebuilt model.
-    * Business card model - Recognize data from business cards using a prebuilt model.
-    * Invoice model - Recognize data from invoices using a prebuilt model.
-    * Identity document model - Recognize data from identity documents using a prebuilt model.
+* Layout - Extract text, table structures, and selection marks, along with their bounding region coordinates, from documents.
```
Comment (samvaity) [inline sdk/formrecognizer/azure-ai-formrecognizer/README.md:6]: Should Extract -> Analyze?

File: sdk/formrecognizer/azure-ai-formrecognizer/README.md [hunk 9]
```diff
@@ -119,49 +130,53 @@ Once completed, set the values of the client ID, tenant ID, and client secret of
...
 
-### FormTrainingClient
-`FormTrainingClient` provides operations for:
+### DocumentModelAdministrationClient
+`DocumentModelAdministrationClient` provides operations for:
 
-- Training custom models without labels to recognize all fields and values found in your custom forms. A `CustomFormModel` is returned indicating the form types the model will recognize, and the fields it will extract for each form type. See the [service documentation][fr-train-without-labels] for a more detailed explanation.
-- Training custom models with labels to recognize specific fields, selection marks, tables, and values you specify by labeling your custom forms. A `CustomFormModel` is returned indicating the fields the model will extract, as well as the estimated accuracy for each field. See the [service documentation][fr-train-with-labels] for a more detailed explanation.
+- Building custom models to extract specific fields you specify by labeling your custom documents. A `DocumentModel` is returned indicating the document type the model will extract, as well as the estimated confidence for each field. See the [service documentation][fr-train-with-labels] for a more detailed explanation.
```
Comment (samvaity) [inline sdk/formrecognizer/azure-ai-formrecognizer/README.md:164]: extracet -> analyze?

Comment (samvaity) [reply sdk/formrecognizer/azure-ai-formrecognizer/README.md:164]: indicating the document type the model will extract -> indicating the document type **of** the model **that** will **analyze**

File: sdk/formrecognizer/azure-ai-formrecognizer/README.md [hunk 10]
```diff
@@ -171,222 +186,264 @@ Sample code snippets are provided to illustrate using long-running operations [b
 
 The following section provides several code snippets covering some of the most common Form Recognizer tasks, including:
 
-* [Recognize Forms Using a Custom Model](#recognize-forms-using-a-custom-model "Recognize Forms Using a Custom Model")
-* [Recognize Content](#recognize-content "Recognize Content")
+* [Extract layout](#extract-layout "Extract Layout")
```
Comment (samvaity) [inline sdk/formrecognizer/azure-ai-formrecognizer/README.md:189]: Analyze Layout ?

File: sdk/formrecognizer/azure-ai-formrecognizer/README.md (hunk 8 above)
Comment (kristapratico) [reply sdk/formrecognizer/azure-ai-formrecognizer/README.md:6]: for layout, I wanted to keep extract since it's does more of just an OCR read of the file.

File: sdk/formrecognizer/azure-ai-formrecognizer/README.md (hunk 10 above)
Comment (kristapratico) [reply sdk/formrecognizer/azure-ai-formrecognizer/README.md:189]: Still prefer to leave this as "extract layout" for OCR and "analyze" for built/prebuilt analysis. Thoughts? @maririos @witemple-msft @catalinaperalta

Comment (maririos) [reply sdk/formrecognizer/azure-ai-formrecognizer/README.md:189]: +1 to the distinction. Also looking at the [comments from Paul](https://github.com/Azure/azure-sdk-for-python/commit/191a9c9d1421cc2b271a078592770c60b3bb6577) I will add `Analyze` to `Document`

Comment (samvaity) [reply sdk/formrecognizer/azure-ai-formrecognizer/README.md:189]: Would it feel awkward to use `Extract` for an API that uses the verbiage `begin**Analyze**Document ?
But I am not too strong on it.

File: sdk/formrecognizer/azure-ai-formrecognizer/README.md [hunk 11]
```diff
@@ -1,15 +1,12 @@
...
-* Custom models - Recognize field values and table data from forms. These models are trained with your own data, so they're tailored to your forms.
-* Content API - Recognize text, table structures, and selection marks, along with their bounding box coordinates, from documents. Corresponds to the REST service's Layout API.
-* Prebuilt models - Recognize data using the following prebuilt models
-    * Receipt model - Recognize data from sales receipts using a prebuilt model.
-    * Business card model - Recognize data from business cards using a prebuilt model.
-    * Invoice model - Recognize data from invoices using a prebuilt model.
-    * Identity document model - Recognize data from identity documents using a prebuilt model.
+* Layout - Extract text, table structures, and selection marks, along with their bounding region coordinates, from documents.
+* Document - Analyze entities, key-value pairs, tables, and selection marks from documents using the general prebuilt document model. 
+* Prebuilt - Analyze data from certain types of common documents (such as receipts, invoices, business cards, or identity documents) using pre-trained models.
```
Comment (samvaity) [inline sdk/formrecognizer/azure-ai-formrecognizer/README.md:8]: nit: pre-trained or prebuilt?

**Review by samvaity — APPROVED** [review]
lgtm!

File: sdk/formrecognizer/azure-ai-formrecognizer/README.md [hunk 12]
```diff
@@ -119,49 +130,53 @@ Once completed, set the values of the client ID, tenant ID, and client secret of
...
 
 ## Key concepts
 
-### FormRecognizerClient
-`FormRecognizerClient` provides operations for:
+### DocumentAnalysisClient
+`DocumentAnalysisClient` provides operations for analyzing input documents using custom and prebuilt models through the `begin_analyze_document` and `begin_analyze_document_from_url` APIs.
+Use the `model` parameter to select the type of model for analysis.
+
+|Model | Features
```
Comment (catalinaperalta) [inline sdk/formrecognizer/azure-ai-formrecognizer/README.md:148]: Tiny nit: does the space after model need to exist?

Comment (catalinaperalta) [reply sdk/formrecognizer/azure-ai-formrecognizer/README.md:148]: Same question for the rows below.

File: sdk/formrecognizer/azure-ai-formrecognizer/README.md [hunk 13]
```diff
@@ -119,49 +130,53 @@ Once completed, set the values of the client ID, tenant ID, and client secret of
...
-    - Invoices. See fields found on an invoice [here][service_recognize_invoice].
-    - Identity documents. See fields found on identity documents [here][service_recognize_identity_documents].
- - Recognizing form content, including tables, lines, words, and selection marks, without the need to train a model. Form content is returned in a collection of `FormPage` objects.
+|"prebuilt-layout" | Text extraction, selection marks, tables
+|"prebuilt-document" | Text extraction, selection marks, tables, key-value pairs and entities
+|"prebuilt-invoices" | Text extraction, selection marks, tables, and pre-trained fields and values pertaining to English invoices
+|"prebuilt-businessCard" | Text extraction and pre-trained fields and values pertaining to English business cards
+|"prebuilt-idDocument" | Text extraction and pre-trained fields and values pertaining to US driver licenses and international passports
+|"prebuilt-receipt" | Text extraction and pre-trained fields and values pertaining to English sales receipts
+|"{custom-model-id}" | Text extraction, selection marks, tables, labeled fields and values from your custom documents
```
Comment (catalinaperalta) [inline sdk/formrecognizer/azure-ai-formrecognizer/README.md:157]: Been seeing the discussion about analyze vs extract in this PR, I agree with the distinction. Just noticing that the other prebuilts like businessCards, idDocs, etc are using extract here. Dont feel too strongly about it, but you might want to update those to use analyze?

File: sdk/formrecognizer/azure-ai-formrecognizer/README.md [hunk 14]
```diff
@@ -171,222 +186,263 @@ Sample code snippets are provided to illustrate using long-running operations [b
...
+            )
+        )
 ```
 
 ### Using Prebuilt Models
-Extract fields from certain types of common forms such as receipts, invoices, business cards, and identity documents using prebuilt models provided by the Form Recognizer service.
+Extract fields from select document types such as receipts, invoices, business cards, and identity documents using prebuilt models provided by the Form Recognizer service.
 
-For example, to extract fields from a sales receipt, use the prebuilt receipt model provided by the `begin_recognize_receipts` method:
+For example, to extract fields from a sales receipt, use the prebuilt receipt model provided by passing `model="prebuilt-receipt"` into the `begin_analyze_documents` method:
```
Comment (catalinaperalta) [inline sdk/formrecognizer/azure-ai-formrecognizer/README.md:274]: Again probably noticing these words a lot more now, should analyze be used here instead of extract?

**Review by catalinaperalta — APPROVED** [review]

File: sdk/formrecognizer/azure-ai-formrecognizer/README.md (hunk 10 above)
Comment (kristapratico) [reply sdk/formrecognizer/azure-ai-formrecognizer/README.md:193]: Going to stick with this for beta.1. we can reevaluate naming again later.

File: sdk/formrecognizer/azure-ai-formrecognizer/README.md (hunk 13 above)
Comment (kristapratico) [reply sdk/formrecognizer/azure-ai-formrecognizer/README.md:161]: pretty much copying service terminology here, I believe text extraction refers to OCR here

File: sdk/formrecognizer/azure-ai-formrecognizer/README.md (hunk 12 above)
Comment (kristapratico) [reply sdk/formrecognizer/azure-ai-formrecognizer/README.md:152]: I can remove, not sure if it affects rendering either way

**Review by maririos — APPROVED** [review]

File: sdk/formrecognizer/azure-ai-formrecognizer/README.md [hunk 15]
```diff
@@ -415,44 +474,9 @@ describes available configurations for retries, logging, transport protocols, an
...
-* Recognize invoices: [sample_recognize_invoices_async.py][sample_recognize_invoices_async]
-* Recognize identity documents: [sample_recognize_identity_documents_async.py][sample_recognize_identity_documents_async]
-* Recognize content: [sample_recognize_content_async.py][sample_recognize_content_async]
-* Recognize custom forms: [sample_recognize_custom_forms_async.py][sample_recognize_custom_forms_async]
-* Train a model without labels: [sample_train_model_without_labels_async.py][sample_train_model_without_labels_async]
-* Train a model with labels: [sample_train_model_with_labels_async.py][sample_train_model_with_labels_async]
-* Manage custom models: [sample_manage_custom_models_async.py][sample_manage_custom_models_async]
-* Copy a model between Form Recognizer resources: [sample_copy_model_async.py][sample_copy_model_async]
-* Create a composed model from a collection of models trained with labels: [sample_create_composed_model_async.py][sample_create_composed_model_async]
+See the [Sample README][sample_readme] for several code snippets illustrating common patterns used in the Form Recognizer Python API.
```
Comment (maririos) [inline sdk/formrecognizer/azure-ai-formrecognizer/README.md:479]: sample readme will be updated in a future PR, right?

Comment (kristapratico) [reply sdk/formrecognizer/azure-ai-formrecognizer/README.md:479]: yep! https://github.com/Azure/azure-sdk-for-python/issues/20841
